internal/pool/pool.go

```diff
@@ -250,7 +255,7 @@ func (p *ConnPool) Get(ctx context.Context) (*Conn, error) {
 		return nil, err
 	}
 
-	for {
+	for i := 0; i < maxBadConnRetries; i++ {
```

Comment:
Let's split this change in a separate PR and leave only checkConn like it was done in https://github.com/go-redis/redis/issues/1821